### PR TITLE
fixes #12425 - do not trigger orchestration during fact import

### PR DESCRIPTION
**Pull request by timogoebel**

### CODE REVIEW

**Comment by timogoebel** [comment]
[test] s seem to have failed randomly

**Comment by timogoebel** [comment]
I think, this should go into 1.11.2 as 1.11.1 made this worse (please see Redmine for details).

File: app/models/host/base.rb [hunk 1]
```diff
@@ -162,7 +168,9 @@ def import_facts(facts)
       # If we don't (e.g. we never install the server via Foreman, we populate the fields from facts
       # TODO: if it was installed by Foreman and there is a mismatch,
       # we should probably send out an alert.
-      save(:validate => false)
+      result = save(:validate => false)
+      self.skip_orchestration = false
```
Comment (dLobatog) [inline app/models/host/base.rb]: I'm thinking you might want to put `self.skip_orchestration = false` on an `ensure` block. 
It's kind of petty but if there's any problem in between lines 161 and this (exceptions etc..), we definitely want to set `skip_orchestration` back to false

File: test/unit/host_test.rb [hunk 2]
```diff
@@ -311,6 +311,13 @@ def teardown
     assert h.import_facts(JSON.parse(File.read(File.expand_path(File.dirname(__FILE__) + "/facts.json")))['facts'])
   end
 
+  test "should not trigger dhcp orchestration when importing facts" do
+    h=Host.new(:name => "sinn1636.lan")
+    h.primary_interface.expects(:dhcp_conflict_detected?).never
+    h.disk = "!" # workaround for now
```
Comment (dLobatog) [inline test/unit/host_test.rb]: Workaround for what?

File: app/models/host/base.rb [hunk 3]
```diff
@@ -74,6 +74,9 @@ def self.taxonomy_conditions
     scope :no_location,     -> { rewhere(:location_id => nil) }
     scope :no_organization, -> { rewhere(:organization_id => nil) }
 
+    # If skip_orchestration is set to true, all orchestration calls will be skipped
+    attr_accessor :skip_orchestration
```
Comment (dLobatog) [inline app/models/host/base.rb]: I would rather call this 'importing_facts' or similar. It's really what it is isn't it? `:someorchestrationtask, :unless => :importing_facts` is more clear about what's going on too IMO. 

We really should move the facts importing logic to a concern :smile:

**Comment by dLobatog** [comment]
:+1: other than that, I have to test it though. I guess just checking that the proxy gets no call at all after importing facts is enough?

**Comment by timogoebel** [comment]
@dLobatog : Followed your naming suggestion, ensured the "reset". Watching the proxy log closely during a fact import is enough to test this.

**Comment by timogoebel** [comment]
[test]

**Comment by dLobatog** [comment]
Merged as 5351c86f9e066fdfae8687e512d35c1fe3e4336f, thanks @timogoebel!

**Pull request closed without merging**